### PR TITLE
ISPN-14138 SQL Store fail to persist numbers with fractional portion

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-14138

### CODE REVIEW

**Comment by wburns** [comment]
@gustavolira Do you think we can test this on the various DB providers?

**Comment by alanfx** [comment]
Hey @wburns and @gustavolira, does the test suite have tests that store and retrieve each kind of data type? Given this issue and the one with Booleans, it seems like it would be good to have!

**Comment by wburns** [comment]
> Hey @wburns and @gustavolira, does the test suite have tests that store and retrieve each kind of data type? Given this issue and the one with Booleans, it seems like it would be good to have!

That is what I have expanded upon with the recent tests. To be honest my problem is I don't know what all the various types a user would use are :( But I can try to find others that we aren't testing, like I know float isn't explicitly tested, but should work.

**Comment by wburns** [comment]
Updated to make sure Numeric with decimals work, added support for Real and Decimal types.

File: persistence/sql/src/main/java/org/infinispan/persistence/sql/AbstractSchemaJdbcStore.java [hunk 1]
```diff
@@ -110,6 +110,9 @@ int typeWeUse(int sqlType, String typeName) {
       } else if (typeName.toUpperCase().startsWith("BOOL")) {
          // Some databases store as int32 or something similar but have the typename as BOOLEAN or some derivation
          return Types.BOOLEAN;
+      } else if (sqlType == Types.NUMERIC && scale == 0) {
+         // If scale is 0 we don't want to use float or double types
+         return Types.INTEGER;
```
Comment (jabolina) [inline persistence/sql/src/main/java/org/infinispan/persistence/sql/AbstractSchemaJdbcStore.java:115]: Just to be on the safe side, should we use a  BIGINT for this case or it is overkill?

**Comment by tristantarrant** [comment]
@wburns conflicts need to be resolved

**Comment by wburns** [comment]
Yeah, I already resolved, was working on changing the types. But I am not having luck, I am going to say for this PR just leave it as is and we can enhance it further later.

**Comment by wburns** [comment]
Rebased.

**Comment by jabolina** [comment]
@wburns Couple of related failures in CI.

**Comment by wburns** [comment]
> @wburns Couple of related failures in CI.

Ah, yes, good catch.. Fixing.

**Comment by wburns** [comment]
Tests should be good now.

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Merged, thanks @wburns

**Pull request merged**